File: algo_basic_python_java/w9.-sorting/problems/6.merge-two_sorted_arrays/merge_two_sorted_arrays.py
class Solution:
    """
    @param A: sorted integer array A
    @param B: sorted integer array B
    @return: A new sorted integer array
    """
    def mergeSortedArray(self, A, B):
        # write your code here
        la, lb = len(A), len(B)
        # pointers to A, B respeactively
        a, b = 0, 0
        res = [0] * (la + lb)
        
        for k in range(la + lb):
            # WARNING!
            # b should >= lb, since when equals the index is already out of range
            if a < la and (b >= lb or A[a] < B[b]):
                res[k] = A[a]
                a+=1
            else:
                res[k] = B[b]
                b+=1
        return res

File: algo_basic_python_java/w9.-sorting/problems/6.merge-two_sorted_arrays/merge_two_sorted_arrays.java
public class Solution {
    /**
     * @param A: sorted integer array A
     * @param B: sorted integer array B
     * @return: A new sorted integer array
     */
    public int[] mergeSortedArray(int[] A, int[] B) {
        // write your code here
        int a = 0, b = 0;
        // pointers to A, B respeactively
        int la = A.length, lb = B.length;
        int[] res = new int[la + lb];

        for (int k = 0; k < la + lb; k++) {
            // WARNING!
            // b should >= lb, since when equals the index is already out of range
            if (a < la && (b >=lb || A[a] < B[b])) {
                res[k] = A[a];
                a++;
            } else {
                res[k] = B[b];
                b++;
            }
        }
        return res;
    }
}
Complexity Analysis